vmdb/db/migrate/20150405141637_remove_port_config_from_container_service.rb

```diff
@@ -0,0 +1,54 @@
+class RemovePortConfigFromContainerService < ActiveRecord::Migration
+  def up
+    create_table :container_service_port_configs do |t|
+      t.string     :ems_ref
+      t.string     :name
+      t.integer    :port
+      t.string     :target_port
+      t.string     :protocol
+      t.belongs_to :container_service, :type => :bigint
+    end
+
+    copy_over_configs
+    remove_columns
+  end
+
+  def copy_over_configs
+    ContainerService.find_each do |service|
+      ContainerServicePortConfig.create(
+          :ems_ref              => "#{service[:ems_ref]}_#{service[:port]}_"\
+                                   "#{service[:container_port]}_#{service[:protocol]}",
+          :port                 => service.port,
+          :protocol             => service.protocol,
+          :target_port          => service.container_port,
+          :container_service_id => service.id
+      )
+    end
+  end
+
+  def add_columns
+    add_column :container_services, :port, :integer
+    add_column :container_services, :protocol, :string
+    add_column :container_services, :container_port, :integer
+  end
+
+  def remove_columns
+    remove_column :container_services, :port
+    remove_column :container_services, :protocol
+    remove_column :container_services, :container_port
+  end
+
+  def down
+    add_columns
```

Comment:
I'm finding the extraction into method makes these migrations more complex.  Can we undo that?  Its pretty uncommon to do that in migrations.